### PR TITLE
bug(TMC-22830) wrong response body is logged in audit logs

### PR DESCRIPTION
**Pull request by ozatsepin**
**What is the problem this Pull Request is trying to solve?** 
wrong response body and code is logged in audit logs in TMC web app

**What is the chosen solution to this problem?**
add response extractor bean to make more flexible mechanism to extract right response code
 
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TMC-22830
 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/ResponseExtractor.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.talend.daikon.spring.audit.logs.service;
+
+public interface ResponseExtractor {
```
Comment (fhuaulme) [inline daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/ResponseExtractor.java:3]: Please add javadoc here

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java [hunk 2]
```diff
@@ -64,10 +66,11 @@ public Object auditLogGeneration(final ProceedingJoinPoint proceedingJoinPoint)
...
             Object auditLogResponseObject = responseObject;
-            if (responseObject instanceof ResponseEntity) {
-                // In case of ResponseEntity, body and status code can be retrieved directly
-                responseCode = ((ResponseEntity) responseObject).getStatusCode().value();
-                auditLogResponseObject = ((ResponseEntity) responseObject).getBody();
+            int code = responseExtractor.getStatusCode(responseObject);
+            if (code != 0) {
+                responseCode = code;
+                // for some responses such 204|302 we should have empty body and retrieve it from response
+                auditLogResponseObject = responseExtractor.getResponseBody(responseObject);
```
Comment (fhuaulme) [inline daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java:73]: Is it ok in your code to have "null" as body content?

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/config/AuditLogAutoConfiguration.java [hunk 3]
```diff
@@ -95,4 +97,22 @@ public AuditLogGeneratorInterceptor auditLogGeneratorInterceptor(AuditLogSender
     public AuditLogIpExtractor auditLogIpExtractor(AuditProperties auditProperties) {
         return new AuditLogIpExtractorImpl(auditProperties);
     }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public ResponseExtractor responseExtractor() {
```
Comment (fhuaulme) [inline daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/config/AuditLogAutoConfiguration.java:103]: What's the purpose of exposing a bean there? Do you expect users of lib to provide their own ResponseExtractor?

**Review by fhuaulme — CHANGES_REQUESTED** [review]

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java (hunk 2 above)
Comment (atelesh) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java:73]: I know POST with empty body looks strange but yes - sometimes it's used

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/config/AuditLogAutoConfiguration.java (hunk 3 above)
Comment (atelesh) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/config/AuditLogAutoConfiguration.java:103]: yes, we still have CXF in some TMC services

Comment (fhuaulme) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/config/AuditLogAutoConfiguration.java:103]: So services using CXF will provide their own ResponseExtractor implementation?

Comment (atelesh) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/config/AuditLogAutoConfiguration.java:103]: exactly

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java (hunk 2 above)
Comment (ozatsepin) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/AuditLogGeneratorAspect.java:73]: yes, before it was ok, and some requests return null body it's expected behavior

File: daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/ResponseExtractor.java (hunk 1 above)
Comment (ozatsepin) [reply daikon-spring/daikon-spring-audit-logs/src/main/java/org/talend/daikon/spring/audit/logs/service/ResponseExtractor.java:6]: updated

**Review by fhuaulme — APPROVED** [review]

**Pull request merged**